clients/vscode-hlasmplugin/src/hlasmLanguageDetection.ts

```diff
@@ -38,14 +38,22 @@ export class HLASMLanguageDetection {
     //automatic detection function
     setHlasmLanguage(document: vscode.TextDocument): boolean {
         // check only plain text files
-        if (document.languageId == 'plaintext') {
+        if (document.languageId == 'plaintext' && !this.EndsWithExtension(document)) {
             if (this.checkHlasmLanguage(document)) {
                 vscode.languages.setTextDocumentLanguage(document, 'hlasm');
                 return true;
             }
         }
         return document.languageId == 'hlasm';
     }
+    //Checks for extension for plaintext files
+    EndsWithExtension(document: vscode.TextDocument): boolean {
+
+        var regexExtension = new RegExp("^[^\\.].*\\.\\w+$");
```

Comment:
```suggestion
        var regexExtension = new RegExp("[^.].*\\.\\w+$");
```
`.hidden` => no ext
`.hidden.cbl` = > `.cbl`?

Reply:
Yes, right .